pywr/parameters/_thresholds.pyx

```diff
@@ -187,6 +187,98 @@ cdef class NodeThresholdParameter(AbstractThresholdParameter):
 NodeThresholdParameter.register()
 
 
+cdef class MultipleThresholdIndexParameter(IndexParameter):
+    """ Multiple threshold holder returning an index based on the previous flow in a node
+    against the given thresholds.
+
+    Parameters
+    ----------
+    node : `pywr.core.AbstractNode`
+    thresholds : iterable of `Parameter` instances or floats
+    """
+    def __init__(self, model, node, thresholds, use_max_flow=False, **kwargs):
+        super(MultipleThresholdIndexParameter, self).__init__(model, **kwargs)
+        self.node = node
+        self.thresholds = thresholds
```

Comment:
If any of these are `Parameter` objects they will need adding as children to this object. Otherwise their calculation order is not guaranteed.

Reply:
I've changed this to load a `ConstantParameter` if not already a parameter and then add children. I took this from the `IndexedArrayParameter` so might not be exactly how the control curve equivalent does it?